middle_end/flambda/terms/code_size.rec.mli

```diff
@@ -0,0 +1,43 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2013--2019 OCamlPro SAS                                    *)
+(*   Copyright 2014--2019 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-4-30-40-41-42"]
+
+(* Computes an approximatation for the code size corresponding to flambda terms.*)
```

Comment:
Maybe also give a formal definition of size as the cost incurred by inlining? Might make some things easier to think about. For instance, why should a closure have the same size as the function it refers to? Because if we copy the closure, we'll make a copy of the function, and the cost of doing so is precisely the function's size.

Reply:
Actually at this point [Code_size] is not so much about the cost incurred by inlining as it aims at being a rough approximation of the size of the generated machine code. I've added a comment in [code_size.rec.ml] about the size of closures as it should be good to have.